### PR TITLE
Upload plugins to a release bucket

### PR DESCRIPTION
**Pull request by rhbuf**
This bucket will be used as a central storage mechanism for plugins.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -37,6 +47,8 @@ jobs:
         run: |
           echo "${MINISIGN_PRIVATE_KEY}" > minisign.key
           go run ./cmd/release -minisign-private-key minisign.key .
+      - name: Upload To Release Bucket
+        run: gcloud storage rsync -r .build/release gs://buf-plugins
```
Comment (pkwarren) [inline .github/workflows/release.yml:51]:
```suggestion
        run: gsutil rsync -r .build/release gs://buf-plugins
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,6 +29,16 @@ jobs:
...
+        # uses https://cloud.google.com/iam/docs/workload-identity-federation to
+        # swap a Github OIDC token for GCP service account credentials, allowing
+        # this workflow to manage the deploy bucket
+      - name: Auth To GCP
+        uses: google-github-actions/auth@v0
+        with:
+          workload_identity_proider: projects/491113660045/locations/global/workloadIdentityPools/plugins-workload-pool/providers/plugins-workload-provider
+          service_account: buf-plugins-1-publisher@buf-plugins-1.iam.gserviceaccount.com
+      - name: Setup GCLOUD
+        uses: 'google-github-actions/setup-gcloud@v0'
```
Comment (pkwarren) [inline .github/workflows/release.yml:41]:
```suggestion
        uses: 'google-github-actions/setup-gcloud@v1'
```

v1 appears to be the latest release - however we're pinning sha hashes for all 3rd party actions in this project.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -29,6 +29,16 @@ jobs:
           go-version: 1.19.x
           check-latest: true
           cache: true
+        # uses https://cloud.google.com/iam/docs/workload-identity-federation to
+        # swap a Github OIDC token for GCP service account credentials, allowing
+        # this workflow to manage the deploy bucket
+      - name: Auth To GCP
+        uses: google-github-actions/auth@v0
```
Comment (pkwarren) [inline .github/workflows/release.yml:36]:
```suggestion
        uses: google-github-actions/auth@v1
```

**Comment by rhbuf** [comment]
superseded by #332

**Pull request closed without merging**